AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -104,9 +104,17 @@ + (void)performDbTokenAsyncOperationWithHttpClient:(id<MSHttpClientProtocol>)htt
             completionHandler([[MSTokensResponse alloc] initWithTokens:nil], serializeError);
             return;
           }
+          if (jsonDictionary[kMSTokens] == nil || [(NSArray *)(jsonDictionary[kMSTokens]) count] == 0) {
```

Comment:
I believe this condition can be combined into one check because Objective-C won't throw when accessing nil object.

```suggestion
          if ([(NSArray *)jsonDictionary[kMSTokens] count] == 0) {
```